Volume/View Models/HomeViewModel.swift

```diff
@@ -0,0 +1,216 @@
+        }
+
+        var networkState: NetworkState?
+        var userData: UserData?
+
+        @Published var trendingArticles: MainView.TabState<[Article]> = .loading
+        @Published var weeklyDebrief: MainView.TabState<WeeklyDebrief?> = .loading
+        @Published var followedArticles: MainView.TabState<[Article]> = .loading
+        @Published var unfollowedArticles: MainView.TabState<[Article]> = .loading
+        @Published var hasMoreFollowedArticlePages = true
+        @Published var hasMoreUnfollowedArticlePages = true
+        @Published var isWeeklyDebriefOpen: Bool = false
+        @Published var deeplinkID: String? = nil
+        @Published var openArticleFromDeeplink: Bool = false
+
+        private var publicationSlugs: MainView.TabState<[String]> = .loading
+        private var queryBag = Set<AnyCancellable>()
+
+        /// Returns empty list if still loading
+        private var loadedPublicationSlugs: [String] {
+            switch publicationSlugs {
+            case .loading:
+                return []
+            case .results(let slugs), .reloading(let slugs):
+                return slugs
+            }
+        }
+
+        func setupEnvironment(networkState: NetworkState, userData: UserData) {
+            self.networkState = networkState
+            self.userData = userData
+        }
+
+        var followedPublicationSlugs: [String] {
+            userData?.followedPublicationSlugs ?? []
+        }
+
+        var hasFollowedPublications: Bool {
+            !followedPublicationSlugs.isEmpty
+        }
+
+        var disableScrolling: Bool {
+            switch trendingArticles {
+            case .loading:
+                return true
+            default:
+                return false
+            }
+        }
+
+        // MARK: Requests
+
+        func fetchContent() {
+            trendingArticles = .loading
+            followedArticles = .loading
+            weeklyDebrief = .loading
+            unfollowedArticles = .loading
+
+            fetchTrendingArticles()
+            fetchWeeklyDebrief()
+            fetchFirstPage()
+        }
+
+        func fetchWeeklyDebrief() {
+            func fetch() {
+                guard let uuid = userData?.uuid else {
+                    #if DEBUG
+                    print("Error: received nil UUID from UserData")
+                    #endif
+                    return
+                }
+
+                Network.shared.publisher(for: GetWeeklyDebriefQuery(uuid: uuid))
+                    .map(\.user?.weeklyDebrief)
+                    .sink { [weak self] completion in
+                        self?.networkState?.handleCompletion(screen: .home, completion)
+                    } receiveValue: { [weak self] weeklyDebriefFields in
+                        guard let weeklyDebriefFields else {
+                            #if DEBUG
+                            print("Error: GetWeeklyDebrief failed on HomeList: field \"weeklyDebrief\" is nil.")
```

Comment:
Since there's no HomeList file anymore maybe this error message should be modified to say "failed on HomeViewModel" instead

Reply:
Good catch